impala/hiveserver2.py

```diff
@@ -942,6 +942,10 @@ def _convert_values(self, type_, is_null, values):
         if type_ == 'DECIMAL':
             for i in range(len(values)):
                 values[i] = (None if is_null[i] else Decimal(values[i]))
+        # CUI-10397 -- fetchcolumnar needs to preserve nulls (not automatically convert to 0)
+        if type_ == 'DOUBLE':
+            for i in range(len(values)):
+                values[i] = (None if is_null[i] else values[i])
```

Comment:
**Question**
I see in the timestamp and decimal types, they parse / convert the values. Should the `else values[i])` here be `else float(values[i]))`

I did see the below comment on a stackoverflow though... so maybe it is better not to convert it?
```
Be aware that if your string number contains more than 15 significant digits float(s) will round it.In those cases it is better to use Decimal
```

Reply:
Yeah, those statements have the two components, the null conversion and the special case type handling. The type conversion is the reason those were added and then the null being converted just happened to happen. RType does this more generally without the type exceptions.

Since types seem to be working otherwise I am not going to change that part.